File: src/Basket.API/Model/CustomerBasket.cs
namespace eShop.Basket.API.Model;

public class CustomerBasket
{
    public string Id { get; set; }

    public string BuyerId { get; set; }

    public List<BasketItem> Items { get; set; } = [];

    public CustomerBasket() { }

    public CustomerBasket(string customerId)
    {
        BuyerId = customerId;
    }
}

File: src-java/eshop-integration-events/src/main/java/com/eshop/integrationevents/CustomerBasket.java
package com.eshop.integrationevents;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.Objects;

public record CustomerBasket(
        @JsonProperty("BuyerId") String buyerId,
        @JsonProperty("Items") List<BasketItem> items) {

    public CustomerBasket {
        Objects.requireNonNull(buyerId, "buyerId");
        items = List.copyOf(Objects.requireNonNull(items, "items"));
    }
}
